gpu/octree/src/cuda/approx_nsearch.cu

```diff
@@ -106,38 +106,112 @@ namespace pcl { namespace device { namespace appnearest_search
 	private:
 
 		__device__ __forceinline__ int findNode()
```

Comment:
This method looks like a copy of the other one in octree_host.cu. If I'm correct in my assessment, we should try to make sure that they're actually using the same function. I'm suggesting this because the code really looks the same.

Reply:
I'm no gonna review this function further because the comments are exactly similar to the other one you already modified.

Reply:
The algorithm is essenitally the same, there are a few subtle differences like the use of the `__popc()` cuda function here.